### PR TITLE
Improve pod autoscaling guide

### PR DESCRIPTION
**Pull request by vlerenc**
**What this PR does / why we need it**:
Improves the pod autoscaling guide.

### CODE REVIEW

**Comment by vlerenc** [comment]
Following our conversation the other day, I clarified a few more things in the pod autoscaling guide @ialidzhikov.

**Review by ScheererJ — COMMENTED** [review]
/lgtm

File: docs/usage/shoot_pod_autoscaling_best_practices.md [hunk 1]
```diff
@@ -199,6 +199,9 @@ There are a few more configurable options of lesser interest:
 
 There are many more options that you can only configure if you deploy your own VPA and which we will not discuss here, but you can check them out [here](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/FAQ.md#what-are-the-parameters-to-vpa-recommender).
 
+> [!NOTE]
+> Due to an implementation detail (smallest bucket size), VPA cannot create recommendations below [10m cores](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/recommender/model/aggregations_config.go#L89-L99) and [10M memory](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/recommender/model/aggregations_config.go#L101-L111) even if `minAllowed` is lower.
```
Comment (vpnachev) [inline docs/usage/shoot_pod_autoscaling_best_practices.md]: Please, use permanent links

```suggestion
> Due to an implementation detail (smallest bucket size), VPA cannot create recommendations below [10m cores](https://github.com/kubernetes/autoscaler/blob/1f89ff92cf87dd3700f74f9b387ae4846aa51846/vertical-pod-autoscaler/pkg/recommender/model/aggregations_config.go#L89-L99) and [10M memory](https://github.com/kubernetes/autoscaler/blob/1f89ff92cf87dd3700f74f9b387ae4846aa51846/vertical-pod-autoscaler/pkg/recommender/model/aggregations_config.go#L101-L111) even if `minAllowed` is lower.
```

Comment (vlerenc) [reply docs/usage/shoot_pod_autoscaling_best_practices.md]: Thank you @vpnachev.

**Comment by vlerenc** [comment]
What needs to happen now for the PR to be merged, please?

**Comment by vlerenc** [comment]
Thank you @rfranzke !

**Pull request merged**